### PR TITLE
Update dockerfiles

### PR DESCRIPTION
**Pull request by yarpryvalikhin**
Dockerfile were added and updated to support Ubuntu 18.04

### CODE REVIEW

File: dockerfiles/Dockerfile.sdl_build.u18.04 [hunk 1]
```diff
@@ -49,4 +49,4 @@ RUN locale-gen en_US.UTF-8
 RUN update-locale LANG=en_US.UTF-8
 ENV LC_ALL en_US.UTF-8
 EXPOSE 22
-CMD ["/usr/sbin/sshd", "-D"]
+CMD ["/usr/sbin/sshd", "-D"]
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04]: I guess we need to add new line here and to other files where it missing

File: dockerfiles/Dockerfile.cppcheck.u18.04 [hunk 2]
```diff
@@ -0,0 +1,40 @@
+FROM ubuntu:18.04
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.cppcheck.u18.04]: I guess it's better to rename dockerfile to `sdl_build.u18.04.cppcheck`

**Comment by dboltovskyi** [comment]
Currently we have the following list of actual docker images:
```
  1. SDL:
    - label: linux_build      => docker_image: sdl_build.u18.04
    - label: linux_coverage   => docker_image: sdl_build.u18.04.cppcheck    
    - label: cppcheck         => docker_image: cppcheck_ubuntu18:latest
    - label: linux_build_u20  => docker_image: sdl_build.u20.04
  2. ATF:
    - label: atf_runner       => docker_image: remote_atf:u18_runner
    - label: remote_atf       => docker_image: remote_atf:u18
    - label: atf_runner_u20   => docker_image: sdl_build.u20.04
  3. Git sync
    - label: git              => docker_image: git:1
```
Please add missing ones

**Comment by dboltovskyi** [comment]
Can we combine `sdl_build.u18.04` and `remote_atf:u18_runner` (and also `remote_atf:u18`) into one image?
E.g. for Ubuntu 20 we have single image for all purposes: `sdl_build.u20.04`

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile.sdl_build.u18.04 (hunk 1 above)
Comment (yarpryvalikhin) [reply dockerfiles/Dockerfile.sdl_build.u18.04]: New line was added

File: dockerfiles/Dockerfile.cppcheck.u18.04 (hunk 2 above)
Comment (yarpryvalikhin) [reply dockerfiles/Dockerfile.cppcheck.u18.04]: Renamed

File: dockerfiles/Dockerfile.sdl_build.u18.04 [hunk 3]
```diff
@@ -1,52 +1,81 @@
...
+        add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" && \
+        apt-get update && \
+        apt-get -yq install docker-ce && \
+        rm -rf /var/cache/apt/archives && \
+        rm -rf /var/lib/apt/lists/*
+RUN echo "mount -t tmpfs tmpfs /home/jenkins" > /entrypoint.sh
+RUN echo "/usr/sbin/sshd -D" >> /entrypoint.sh
 EXPOSE 22
-CMD ["/usr/sbin/sshd", "-D"]
+CMD ["/entrypoint.sh"]
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04]: When I run a container for this image 
```
docker run --rm -it c046fe18a3df
```
I got the following message:
```
docker: Error response from daemon: OCI runtime create failed: container_linux.go:370: starting container process caused: exec: "/entrypoint.sh": permission denied: unknown.
```

However on Jenkins through `ssh` it works fine

File: dockerfiles/Dockerfile.sdl_build.u18.04 [hunk 4]
```diff
@@ -1,52 +1,81 @@
...
+        python3-setuptools \
+        software-properties-common \
+        openssh-server \
+        && python3 -m pip install xmlschema
+RUN apt-get update && apt-get -yq install \
+        software-properties-common \
+        openssl \
+        liblua5.2-0 \
+        liblua5.2-dev \
+        lua-lpeg-dev \
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04]: Package `lua5.2` is missing. `ATF` can't be built without it. See details: http://172.26.147.12/view/Dockerfiles/job/Dockerfiles_u18_ATF_BUILD_RUNNER/1/console

File: dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    libssl-dev libudev-dev
+
+RUN add-apt-repository --yes ppa:openjdk-r/ppa && \
+    apt-get -qq update && \
+    apt-get install -y default-jre
+
+#RUN sed -i 's|session required pam_loginuid.so|session optional pam_loginuid.so|g' /etc/pam.d/sshd
+#RUN mkdir -p /var/run/sshd
+#RUN adduser --quiet jenkins && echo "jenkins:jenkins" | chpasswd && \
+#    sudo echo "jenkins ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck]: Are these lines commented intentionally? `sshd` is required to start container within Jenkins.
I was not able to start this container and got error:
```
Note: Disabled 4 min 37 sec ago due to error. Will re-enable in 22 sec.
Reason: Template provisioning failed.

java.io.IOException: SSH service hadn't started after 20 seconds.	at io.jenkins.docker.connector.DockerComputerSSHConnector.createLauncher(DockerComputerSSHConnector.java:256)
```

File: dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+#RUN apt-get -yq update && \
+#    apt install -yq gdb sqlite locales clang-format-3.6\
+#    libbluetooth3 libbluetooth-dev libavahi-client-dev bluez-tools \
+#    sqlite3 libsqlite3-dev libexpat1-dev qt5-default libqt5websockets5-dev libusb-1.0-0-dev \
+#    lcov genhtml html2text bc xmlstarlet xsltproc dos2unix
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck]: Don't we need all these packages?

File: dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+RUN apt install cppcheck -y
+
+#RUN locale-gen en_US.UTF-8
+#RUN update-locale LANG=en_US.UTF-8
+#ENV LC_ALL en_US.UTF-8
+
+EXPOSE 22
+CMD ["/usr/sbin/sshd", "-D"]
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck:52]: Does an image created within this Dockerfile allow to run `linux_coverage` as well as `cppcheck` jobs?

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck (hunk 7 above)
Comment (dboltovskyi) [reply dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck:52]: Looks like no. Please find details: http://172.26.147.12/view/Dockerfiles/job/Dockerfiles_u18_SDL_Coverage/

File: dockerfiles/entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (dboltovskyi) [inline dockerfiles/entrypoint.sh]: Why do we need this entrypoint?
E.g. for Dockerfile.sdl_build.u20.04 it is not required

File: dockerfiles/entrypoint.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+chown developer /home/developer
+chgrp developer /home/developer
+
+export HOME=/home/developer
+
+export LANG=en_US.UTF-8
+cd /home/developer/sdl/sdl_atf/build/bin/RemoteTestingAdapterServer
+ifconfig
+sudo LD_LIBRARY_PATH="/home/developer/sdl/3rd_party/lib:/home/developer/sdl/3rd_party/x86_64/lib:." -u developer "./RemoteTestingAdapterServer"
```
Comment (dboltovskyi) [inline dockerfiles/entrypoint.sh]: Starting of `RemoteTestingAdapterServer` is implemented in https://raw.githubusercontent.com/LuxoftSDL/sdl_core_ci/impl/atf_remote_upd/scripts/atf_runner_remote.sh which is used in test jobs for remote connection:
 - Dockerfiles_u18_ATF_Smoke_TCP_Remote_P
 - Dockerfiles_u20_ATF_Smoke_TCP_Remote_P

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: dockerfiles/Dockerfile.sdl_build.u18.04 [hunk 10]
```diff
@@ -1,52 +1,52 @@
...
+  && rm -rf /var/lib/apt/lists/*
+
+# ATF run dependencies
+RUN apt-get update && apt-get -q -y install net-tools iproute2 rsync screen gdb
+
+# ATF build dependencies
+RUN apt-get update && apt-get -q -y install qt5-default libqt5websockets5-dev liblua5.2-dev libxml2-dev lua5.2
+
+# SDL build dependencies
+RUN apt-get update && apt-get -q -y install cmake libssl-dev libusb-1.0-0-dev \
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04:28]: @yarpryvalikhin Please notice within image created from this version of Dockerfile it's impossible to build ATF.
Details in http://172.26.147.12/view/Dockerfiles/job/Dockerfiles_u18_ATF_BUILD_RUNNER/5/console
The reason of that is absence of `cmake-3.11` which was removed in latest commit https://github.com/LuxoftSDL/sdl_core_ci/pull/4/commits/9d7f261193bb5428dee02d27db43619cf4a91fac

File: dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+RUN sed -i 's|session required pam_loginuid.so|session optional pam_loginuid.so|g' /etc/pam.d/sshd
+RUN mkdir -p /var/run/sshd
+RUN adduser --quiet jenkins && echo "jenkins:jenkins" | chpasswd && \
+    sudo echo "jenkins ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+RUN locale-gen en_US.UTF-8
+RUN update-locale LANG=en_US.UTF-8
+ENV LC_ALL en_US.UTF-8
+EXPOSE 22
+CMD ["/usr/sbin/sshd", "-D"]
+
```
Comment (dboltovskyi) [inline dockerfiles/Dockerfile.sdl_build.u18.04.cppcheck]: New line is missing at the end of the file.

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

**Review by dboltovskyi — APPROVED** [review]
@yarpryvalikhin Thank you for updates

**Pull request merged**